markdown/events/2022/bytesize-36-multiqc.md

```diff
@@ -24,3 +24,119 @@ It is our hope that these talks / videos will build an archive of training mater
+
+[3:07](https://www.youtube.com/watch?v=a3_IYSMrKAk&t=187)
+Bioinformatics tools are famous for lacking standards in file formats. MultiQC does some of that legwork for you and you it gives you tab separated files by default, but you can have YAML or JSON as well. All the different 115 bioinformatics tools will produce output, which is in roughly the same flavor. It's useful for downstream processing as well. MultiQC is written with Python, so it's pretty easy to install if you've got Python set up on your system using a Python package index here. `pip install multiqc`. It's also in Conda, or you can use it with Docker or Singularity, there's a Galaxy wrapper before it. Most places you're already running software, you'll find MultiQC there. There's a Debian installation and all sorts.
+
+[3:55](https://www.youtube.com/watch?v=a3_IYSMrKAk&t=235)
+To run MultiQC, you call the MultiQC command, which is the tool name. It needs a minimum of just one argument, which is a file path. In this case, I've given it a dot, which just means the current working directory in the terminal. MultiQC then will recursively look through every file and folder in that path and see what it can find. Anything that's not a log file, or that it doesn't recognize, it will just ignore. It's been designed from the ground up to work with analysis pipelines, where you have all of your results in a folder, and then you just run `multiqc [folder]`, and it will find what's relevant. If you want, it can also take explicit file names and as many different paths as you want, if that's better for your setup. That's it. Once you've run MultiQC, it will tell you it's generated an HTML report, and then it's up to you, the human, to do the difficult bit, which is to look at that report, understand what it's telling you, and continue with your analysis.
+
+[4:50](https://www.youtube.com/watch?v=a3_IYSMrKAk&t=290)
+I started MultiQC back in 2016, and it has wildly exceeded my expectations. I was looking this up yesterday. If any of you follow me on Twitter, you might have seen, it just passed 2000 citations now, paper citations for MultiQC paper, which is just utterly mind-blowing. I certainly didn't set out with any expectations of this. It was just an internal tool that we needed at SciLifeLab for our own internal QC. It's very humbling that MultiQC has reached and helped so many people. There you go, 114 different bioinformatics tools supported, more coming in all the time. Those citations, I find quite terrifying, if I'm honest. You can see I've got this going up and up, and it makes me always very scared to push a new release, because I always think there's always people using it. What happens if I've broken something? Or worse, what happens if I find out that something has been broken for the past three years and all these citations are wrong? But anyway, that's for maintainers' nightmare.
```

Comment:
```suggestion
I started MultiQC back in 2016, and it has wildly exceeded my expectations. I was looking this up yesterday. If any of you follow me on Twitter, you might have seen, it just passed 2000 citations now, paper citations for MultiQC paper, which is just utterly mind-blowing. I certainly didn't set out with any expectations of this. It was just an internal tool that we needed at SciLifeLab for our own internal QC. It's very humbling that MultiQC has reached and helped so many people. There you go, 114 different bioinformatics tools supported, more coming in all the time. Those citations, I find quite terrifying, if I'm honest. You can see the graph is going up and up, and it makes me always very scared to push a new release, because I always think there's always people using it. What happens if I've broken something? Or worse, what happens if I find out that something has been broken for the past three years and all these citations are wrong? But anyway, that's for maintainers' nightmare.
```